### DISCUSSION TITLE
Cannot change fan icon in Home app

### DISCUSSION THREAD

**Discussion by blaisebarrette** [HomeSpan/HomeSpan#380] [General]
I successfully made a ceiling fan control and it’s been rock steady for about two months now. Very pleased with the result!
A very small and unimportant detail… I can’t modify the icon in the Home app. Since it’s a ceiling fan, it would be nice if I could change the icon to reflect the type of fan. Any idea why that is?

Thanks!

**Comment by HomeSpan** [comment]
I don't think there is a way to change the fan icon.  The Home App seems to only allow customization of a light bulb device.  I have a commercial HomeKit-enabled ceiling fan that indeed shows a ceiling fan icon, suggesting that the commercial version of HAP has some additional features that the non-commercial version does not have.

**Reply by josephcherayil** [reply]
instead of fan try making it a switch, it will give 3 options switch,bulb and fan in the fan icon there are 3 options and ceiling fan is there.

**Reply by tn-cs** [reply]
> instead of fan try making it a switch, it will give 3 options switch,bulb and fan in the fan icon there are 3 options and ceiling fan is there.

struct HAP_Switch:Service::Switch{}

![IMG_2866](https://github.com/HomeSpan/HomeSpan/assets/27691731/6f8841e5-5851-417f-9ad0-cab206341e9a)

struct HAP_Fan:Service::Fan{}

![IMG_2864](https://github.com/HomeSpan/HomeSpan/assets/27691731/3656203c-6058-4d3e-a407-b6d93a6e10fc)

Click icon's
![IMG_2867](https://github.com/HomeSpan/HomeSpan/assets/27691731/d68d4360-3704-4f35-a567-8dcf0f000e46)

**It works with the library HomeSpan!!!!!!!!!!!!!**

**Comment by blaisebarrette** [comment]
Oh! Ok! Thanks! I didn’t realized there are commercial and non-commercial versions.
Love your library! Thanks again!

- Blaise

**Comment by danilkorotkov** [comment]
HomeSpan
![Снимок экрана 2023—04—07 в 10 23 57](https://user-images.githubusercontent.com/25537053/230562516-be590e01-7373-4541-a894-2d32ca6d24ba.jpeg)

**Reply by josephcherayil** [reply]
Yeah not these three are available

**Reply by danilkorotkov** [reply]
Because your accessory is not fan service, but switch

**Comment by tn-cs** [comment]
Here's what I noticed,

I only used iPhone SE 2020 and custom device with HomeSpan library,
cannot change icon:

![IMG_2528](https://github.com/HomeSpan/HomeSpan/assets/27691731/b7d830b8-1f4f-42e7-b992-40a734fab72a)
![IMG_2527](https://github.com/HomeSpan/HomeSpan/assets/27691731/c5100adf-8b06-43d0-b494-1e32e40a4b1d)


**Then custom device with HomeSpan library and iPhone 11 and HomePod mini:**

![IMG_2862](https://github.com/HomeSpan/HomeSpan/assets/27691731/50c8c8eb-7d6e-4aa7-83f1-6d045f4914f4)
![IMG_2863](https://github.com/HomeSpan/HomeSpan/assets/27691731/84a1c7e4-295e-42c3-ba11-05efb0042595)
